modules/service/src/main/resources/db/migration/V0999__gmos_imaging_modes.sql

```diff
@@ -0,0 +1,163 @@
+-- GMOS North Imaging Mode table
+CREATE TABLE t_gmos_north_imaging (
+  c_observation_id       d_observation_id    NOT NULL REFERENCES t_observation(c_observation_id) ON DELETE CASCADE,
+  c_observing_mode_type  e_observing_mode_type NOT NULL DEFAULT 'gmos_north_imaging'::e_observing_mode_type,
+  c_instrument           d_tag NOT NULL DEFAULT 'GmosNorth',
+
+  -- Explicit overrides for defaults (can be NULL to use computed defaults)
+  c_explicit_bin           d_tag             NULL REFERENCES t_gmos_binning(c_tag),
+  c_explicit_amp_read_mode d_tag             NULL REFERENCES t_gmos_amp_read_mode(c_tag),
+  c_explicit_amp_gain      d_tag             NULL REFERENCES t_gmos_amp_gain(c_tag),
+  c_explicit_roi           d_tag             NULL REFERENCES t_gmos_roi(c_tag),
+
+  PRIMARY KEY (c_observation_id, c_observing_mode_type),
+  FOREIGN KEY (c_observation_id, c_instrument, c_observing_mode_type) REFERENCES t_observation(c_observation_id, c_instrument, c_observing_mode_type) ON DELETE CASCADE DEFERRABLE INITIALLY DEFERRED,
+  
+  CHECK (c_observing_mode_type = 'gmos_north_imaging'),
+  CHECK (c_instrument = 'GmosNorth')
+);
+
+COMMENT ON TABLE t_gmos_north_imaging IS 'GMOS North Imaging mode configuration';
+
+-- GMOS South Imaging Mode table
+CREATE TABLE t_gmos_south_imaging (
+  c_observation_id       d_observation_id    NOT NULL REFERENCES t_observation(c_observation_id) ON DELETE CASCADE,
+  c_observing_mode_type  e_observing_mode_type NOT NULL DEFAULT 'gmos_south_imaging'::e_observing_mode_type,
+  c_instrument           d_tag NOT NULL DEFAULT 'GmosSouth',
+
+  -- Explicit overrides for defaults (can be NULL to use computed defaults)
+  c_explicit_bin           d_tag             NULL REFERENCES t_gmos_binning(c_tag),
+  c_explicit_amp_read_mode d_tag             NULL REFERENCES t_gmos_amp_read_mode(c_tag),
+  c_explicit_amp_gain      d_tag             NULL REFERENCES t_gmos_amp_gain(c_tag),
+  c_explicit_roi           d_tag             NULL REFERENCES t_gmos_roi(c_tag),
+
+  PRIMARY KEY (c_observation_id, c_observing_mode_type),
+  FOREIGN KEY (c_observation_id, c_instrument, c_observing_mode_type) REFERENCES t_observation(c_observation_id, c_instrument, c_observing_mode_type) ON DELETE CASCADE DEFERRABLE INITIALLY DEFERRED,
+
+  CHECK (c_observing_mode_type = 'gmos_south_imaging'),
+  CHECK (c_instrument = 'GmosSouth')
+);
+
+COMMENT ON TABLE t_gmos_south_imaging IS 'GMOS South Imaging mode configuration';
+
+-- Normalized tables for filter arrays following the d_tag array pattern
+
+-- GMOS North Imaging Filters - normalized table for array support
+CREATE TABLE t_gmos_north_imaging_filter (
+  c_observation_id d_observation_id NOT NULL,
+  c_filter         d_tag            NOT NULL REFERENCES t_gmos_north_filter(c_tag),
+  PRIMARY KEY (c_observation_id, c_filter),
+  FOREIGN KEY (c_observation_id) REFERENCES t_observation(c_observation_id) ON DELETE CASCADE
```

Comment:
I wonder if this should instead be a FK constraint to `t_gmos_north_imaging(c_observation_id)`?  We presumably don't want values hanging out in this filter table if the mode switches and there is no longer an entry for the observation in `t_gmos_north_imaging`.

Reply:
Good point, I'll implement that